### PR TITLE
Add guide on JS modules

### PR DESCRIPTION
**Pull request by mathiasbynens**
**Target Live Date:** 2018-06-18

- [x] This has been reviewed and approved by the team
- [x] I have run `gulp test` locally and all tests pass.
- [x] I have added the appropriate `type-something` label.
- [x] I've staged the site and manually verified that my content displays correctly.

### CODE REVIEW

**Comment by mathiasbynens** [comment]
@WebFundBot All those are false positives.

File: src/content/en/fundamentals/primers/modules.md [hunk 1]
```diff
@@ -0,0 +1,578 @@
...
+The browser knows it’s a module because of the `type` attribute on the script element.
+
+Still, we recommend using the `.mjs` extension for modules, for two reasons:
+
+1. During development, it makes it crystal clear that the file is a module as opposed to a
+    regular script. (It’s not always possible to tell just by looking at the code.) As mentioned
+    before, modules are treated differently than regular scripts, so the difference is hugely
+    important!
+2. It’s consistent with Node.js, where [the experimental modules implementation]
+    (https://nodejs.org/api/esm.html) only supports files with the `.mjs` extension.
```
Comment (petele) [inline src/content/en/fundamentals/primers/modules.md]: you can't have breaklines in links, the `)[` have to be on the same line.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -7,7 +7,7 @@ What's changed, or what was fixed?
 **Target Live Date:** YYYY-MM-DD
 
 - [ ] This has been reviewed and approved by (NAME)
-- [ ] I have run `gulp test` locally and all tests pass.
+- [ ] I have run `npm test` locally and all tests pass.
```
Comment (mathiasbynens) [inline .github/PULL_REQUEST_TEMPLATE.md]: Drive-by improvement: `npm test` still works even if `gulp` is not installed globally.

File: src/content/en/fundamentals/primers/modules.md [hunk 3]
```diff
@@ -0,0 +1,578 @@
...
+modules](https://docs.google.com/document/d/1ovo4PurT_1K4WFwN2MYmmgbLcr7v6DRQN67ESVA-wq0/pub), the
+loading performance of bundled applications is better than unbundled ones.
+
+[![](imgs/renderer-main-thread-time-breakdown.png)](https://docs.google.com/document/d/1ovo4PurT_1K4WFwN2MYmmgbLcr7v6DRQN67ESVA-wq0/pub)
+
+One reason for this is that the static `import`/`export` syntax is statically analyzable, and it
+can thus help bundler tools optimize your code by eliminating unused exports. Static `import` and
+`export` are more than just syntax; they are a critical tooling feature!
+
+**Our general recommendation is to continue using bundlers before deploying modules to
```
Comment (petele) [inline src/content/en/fundamentals/primers/modules.md]: Total nit - you could use `Success: We recommend you use bundlers...` Success is a special callout similar to note that marks the whole block green with a checkmark to the right

File: src/content/en/fundamentals/primers/modules.md (hunk 1 above)
Comment (mathiasbynens) [reply src/content/en/fundamentals/primers/modules.md]: Good catch! Fixed.

File: src/content/en/fundamentals/primers/modules.md (hunk 3 above)
Comment (mathiasbynens) [reply src/content/en/fundamentals/primers/modules.md]: Good call! Done.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 2 above)
Comment (petele) [reply .github/PULL_REQUEST_TEMPLATE.md]: 👍
